Gems/AWSCore/Code/Source/Editor/UI/AWSCoreResourceMappingToolAction.cpp

```diff
@@ -11,124 +11,141 @@
 #include <AzFramework/StringFunc/StringFunc.h>
 
 #include <AWSCoreInternalBus.h>
+#include <Configuration/AWSCoreConfiguration.h>
 #include <Editor/UI/AWSCoreResourceMappingToolAction.h>
 
 namespace AWSCore
 {
-    AWSCoreResourceMappingToolAction::AWSCoreResourceMappingToolAction(const QString& text)
-        : QAction(text)
+    AWSCoreResourceMappingToolAction::AWSCoreResourceMappingToolAction(const QString& text, QObject* parent)
+        : QAction(text, parent)
         , m_isDebug(false)
         , m_enginePythonEntryPath("")
         , m_toolScriptPath("")
         , m_toolQtBinDirectoryPath("")
-        , m_toolLogPath("")
+        , m_toolLogDirectoryPath("")
+        , m_toolConfigDirectoryPath("")
         , m_toolReadMePath("")
     {
-        auto engineRootPath = AZ::IO::FileIOBase::GetInstance()->GetAlias("@engroot@");
-        if (!engineRootPath)
+        InitAWSCoreResourceMappingToolAction();
+    }
+
+    void AWSCoreResourceMappingToolAction::InitAWSCoreResourceMappingToolAction()
```

Comment:
This entire function can be simplified by using AZ::IO::Path class and the AzCore Utils.h header

Reply:
For example
```
AZ::IO::Path engineRootPath{ AZ::IO::PathView(AZ::Utils::GetEnginePath()) };
AZ::IO::Path projectPath{ AZ::IO::PathView(AZ::Utils::GetProjectPath()) };

m_enginePythonEntryPath = engineRootPath/ EngineWindowsPythonEntryScriptPath;
m_toolScriptPath = engineRootPath / ResourceMappingToolDirectoryPath / "resource_mapping_tool.py";
m_toolReadMePath = engineRootPath / ResourceMappingToolDirectoryPath / "README.md";

m_toolLogDirectoryPath = projectPath / ResourceMappingToolLogDirectoryPath;
m_toolConfigDirectoryPath = projectPath / AWSCoreConfiguration::AWSCoreResourceMappingConfigFolderName;
```